### PR TITLE
add workflow azure-ipi-private-sdn and gcp-upi-private-xpn-sdn

### PR DESCRIPTION
**Pull request by lihongan**

### CODE REVIEW

**Comment by jianli-wei** [comment]
@lihongan Forgive me, I'm unfamiliar with SDN. Two questions, thanks in advance! 

1. `oc get networks.config.openshift.io cluster -oyaml` (in `cucushift-installer-check` [build logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/31929/rehearse-31929-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-gcp/1565550174380494848/artifacts/installer-rehearse-gcp/cucushift-installer-check/build-log.txt)) tells `networkType: OVNKubernetes`, is it expected?  
2. It's said the default value for `networking.networkType` is `OpenShiftSDN` (e.g. [4.11 doc](https://docs.openshift.com/container-platform/4.11/installing/installing_gcp/installing-gcp-customizations.html#installation-configuration-parameters-network_installing-gcp-customizations)). Why do we need to set it explicitly?

**Comment by lihongan** [comment]
> @lihongan Forgive me, I'm unfamiliar with SDN. Two questions, thanks in advance!
> 
>     1. `oc get networks.config.openshift.io cluster -oyaml` (in `cucushift-installer-check` [build logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/31929/rehearse-31929-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-gcp/1565550174380494848/artifacts/installer-rehearse-gcp/cucushift-installer-check/build-log.txt)) tells `networkType: OVNKubernetes`, is it expected?
> 
>     2. It's said the default value for `networking.networkType` is `OpenShiftSDN` (e.g. [4.11 doc](https://docs.openshift.com/container-platform/4.11/installing/installing_gcp/installing-gcp-customizations.html#installation-configuration-parameters-network_installing-gcp-customizations)). Why do we need to set it explicitly?
> 
> 
> BTW, please give some description for the PR, thanks!


@jianli-wei These workflows are for 4.12 and later version since from 4.12 the default value is OVN, more details please see https://issues.redhat.com/browse/OCPQE-10828

And for the 1st question, it is not expected and I need a further investigation. Thanks.

**Comment by lihongan** [comment]
for the update of vsphere part, it is from https://github.com/openshift/release/pull/31841#discussion_r960227719

**Comment by lihongan** [comment]
/assign @jianlinliu 
All look good now and SDN is set as expected.

cc @zhaozhanqi @anuragthehatter

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/upi/private/xpn/sdn/provision/cucushift-installer-rehearse-gcp-upi-private-xpn-sdn-provision-chain.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  - ref: ipi-install-rbac
+  - ref: upi-conf-gcp-xpn
+  - ref: ignition-bastionhost
+  - ref: gcp-provision-bastionhost
+  - ref: proxy-config-generate
+  - chain: ipi-conf-gcp
+  - ref: sdn-conf
+  - ref: upi-conf-gcp
+  - ref: upi-install-gcp
+  - ref: ipi-install-times-collection
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/gcp/upi/private/xpn/sdn/provision/cucushift-installer-rehearse-gcp-upi-private-xpn-sdn-provision-chain.yaml:13]: Can  you help add `- ref: enable-qe-catalogsource` here.

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/gcp/upi/private/xpn/sdn/provision/cucushift-installer-rehearse-gcp-upi-private-xpn-sdn-provision-chain.yaml:13]: @jianli-wei I just checked several `cucushift-installer-rehearse-gcp-upi-private-xpn-provision`, did not append `- ref: enable-qe-catalogsource` there, can you help check if the existing workflow missing them, and file another PR to fix them.
The rules is connected env (also including behind proxy env) need `enable-qe-catalogsource` step, disconnected env need `enable-qe-catalogsource-disconnected` and `set-sample-operator-disconnected` step.

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/upi/private/xpn/sdn/provision/cucushift-installer-rehearse-gcp-upi-private-xpn-sdn-provision-chain.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  - ref: enable-qe-catalogsource
```
Comment (lihongan) [inline ci-operator/step-registry/cucushift/installer/rehearse/gcp/upi/private/xpn/sdn/provision/cucushift-installer-rehearse-gcp-upi-private-xpn-sdn-provision-chain.yaml:14]: Sure, `ref: enable-qe-catalogsource` is added.

**Comment by lihongan** [comment]
azure job failed 
```
level=error msg=failed to initialize the cluster: Cluster operators authentication, console, dns, image-registry, ingress, machine-api, monitoring are not available
```

**Pull request merged**